packages/medusa/src/interfaces/batch-job-strategy.ts

```diff
@@ -0,0 +1,71 @@
+import { EntityManager } from "typeorm"
+import { BatchJob } from "../models/batch-job"
+
+export interface IBatchJobStrategy {
+  /**
+   * Instantiate a new price selection strategy with the active transaction in
+   * order to ensure reads are accurate.
+   * @param manager EntityManager with the queryrunner of the active transaction
+   * @returns a new price selection strategy
+   */
+  withTransaction(manager: EntityManager): IBatchJobStrategy
+
+  /*
+   * Used in the API controller to verify that the `context` param is valid
+   */
+  validateContext(
+    context: Record<string, unknown>
+  ): Promise<Record<string, unknown>>
+
+  /*
+   *  Method does the actual processing of the job. Should report back on the progress of the operation.
+   */
+  processJob(batchJobId: string): Promise<BatchJob>
+
+  /*
+   *  Method performs the completion of the job. Will not be run if `processJob` has already moved the BatchJob to a `complete` status.
+   */
+  completeJob(batchJobId: string): Promise<BatchJob>
+
+  /**
+   * Validates that the file can be used for processJob
+   */
+  validateFile(fileLocation: string)
+
+  /**
+   * Builds and returns a template file that can be downloaded and filled in
+   */
+  buildTemplate()
+}
+
+export abstract class AbstractBatchJobStrategy implements IBatchJobStrategy {
+  static identifier: string
```

Comment:
you could do that instead to enforce it

```ts
  public getIdentifier(): string {
    if (!(<typeof AbstractBatchJobStrategy>this.constructor).identifier) {
      throw new Error('Missing static property "identifier".')
    }
    return (<typeof AbstractBatchJobStrategy>this.constructor).identifier
  }
```

Or static if this is on the class that we check

Reply:
This doesn't work due to the way we extend the transaction service and implement the abstractBatchJobStrategy. An alternative would be to create abstract getters that force the implementation an instantiating the strategy with a `new` keyword. Then we can use the methods to obtain identifier and batchtype getters like this: `new module({}, pluginDetails.options).getIdentifier` it just feels a bit hacky

Reply:
Oh yeah sure, if it is to be used statically yoi don’t need that👍